src/tensor_ops/sum_to/sum_to.cu

```diff
@@ -0,0 +1,65 @@
+__device__ unsigned int get_strided_index(
+    unsigned int idx,
+    size_t num_dims,
+    const size_t *dims,
+    const size_t *strides
+) {
+    unsigned int strided_i = 0;
+    for (unsigned int d = 0; d < num_dims; d++) {
+        unsigned int dim_idx = num_dims - 1 - d;
+        strided_i += (idx % dims[dim_idx]) * strides[dim_idx];
+        idx /= dims[dim_idx];
+    }
+    return strided_i;
+}
+
+// Accepts pre-broadcasted strides for both input & output.
+// So both inp & out are expected to be broadcasted to the same size.
+extern "C" __global__ void sum_to_forward(
+    const size_t numel,
+    const size_t num_dims,
+    const size_t *dims,
+    const float *inp,
+    const size_t *inp_strides,
+    float *out,
+    const size_t *out_strides
+) {
+    unsigned int i = blockIdx.x * blockDim.x + threadIdx.x;
+
+    if (i >= numel) {
+        return;
+    }
+
+    unsigned int inp_strided_i = get_strided_index(i, num_dims, dims, inp_strides);
+    auto tmp = inp[inp_strided_i];
+
+    unsigned int out_strided_i = get_strided_index(i, num_dims, dims, out_strides);
+    atomicAdd(out + out_strided_i, tmp);
```

Comment:
While this is okay to start with, this will be really bad performance when reducing large tensors to 1 element. Need to add an issue to create a special kernel for that case (and there are standard ways to write high performant ones of those)